packages/medusa/src/services/payment-collection.ts

```diff
@@ -401,7 +401,9 @@ export default class PaymentCollectionService extends TransactionBaseService {
       }
 
       let authorizedAmount = 0
-      for (const session of payCol.payment_sessions) {
+      for (let i = 0; i < payCol.payment_sessions.length; i++) {
+        const session = payCol.payment_sessions[i]
+
```

Comment:
thought: looks like we are authorizing all payment sessions here? Can you elaborate a bit on how the flow is? For comparison the way payment sessions work with cart right now is:

1. PaymentSessions are created for each of the payment providers enabled in the cart's region
2. Customer selects which payment provider they wish to use e.g. Stripe
3. Customer clicks pay
4. Only the Stripe payment session is authorized

With the approach here it looks like we expect the customer to authorize all payment sessions - alternatively, we will have to make sure that only the customer's selected payment sessions are in `payCol.payment_sessions`?

Reply:
The difference from the current Cart flow is that we only initialize the sessions that the client selected. When they change the selection the service will always manage to initialize the selected session and cancel/delete the others. Same for multiple sessions.
The endpoint to manage sessions always receive the selected sessions, the rest of the flow is the same.